feat: CSV File Upload form updates

Antonio-RiveroMartnez opened the pull request:
### SUMMARY
This is going to be the first of a set of PRs to improve the experience for our users in the CSV file upload form. Here we are reordering some fields of the form and changing their Name and descriptions.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
https://user-images.githubusercontent.com/38889534/198689750-2996d449-af67-43f6-b75f-94c79c913960.mov

After:
https://user-images.githubusercontent.com/38889534/198690113-03a711d0-f903-484e-a844-cae9fda8a8eb.mov



### TESTING INSTRUCTIONS
1. Open the CSV file upload form
2. Check the form matches the proposed changes

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [X] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Antonio-RiveroMartnez committed:
CSV Upload:

- Update field texts: Name & Descriptions
- Reorder form fields
- Update tests

eschutho commented:
/testenv up

eschutho commented on superset/views/database/forms.py:
```diff
@@ -133,129 +124,136 @@ class CsvToDatabaseForm(UploadToDatabaseForm):
+        description=_(
+            "Json list of the values that should be treated as null. "
+            'Examples: [""], ["None", "N/A"], ["nan", "null"]. '
+            "Warning: Hive database supports only single value"
```
nit typo in existing text.. should this be "only a single value"?

Antonio-RiveroMartnez replied:
Oh Thanks! Fixed

Antonio-RiveroMartnez committed:
CSV Upload:

- Fix typo in Null Values field

eschutho reviewed (approved):
Code looks great. Will defer to @jinghua-qa or @yousoph for visual review.

Antonio-RiveroMartnez committed:
CSV Upload:

- Make description for Null Values field match the mock up

Antonio-RiveroMartnez committed:
CSV Upload:

- Rename the id of the DOm element so it doesn't fail when querying schemas with allowed csv file upload

jinghua-qa commented:
/testenv up

Antonio-RiveroMartnez committed:
CSV Upload:

- The DOM element was still misnamed in one place

andrey-zayats commented:
LGTM!
